source/exec/iexecprovider.d

```diff
@@ -8,5 +8,5 @@ import std.typecons: Tuple;
 +/
 interface IExecProvider
 {
-	Tuple!(string, "output", bool, "success") compileAndExecute(string source);
+	Tuple!(string, "output", bool, "success") compileAndExecute(string source, string compiler = "dmd");
```

Comment:
Or alternatively, you could add a test that verifies that passing random strings is well handled.
Edit:
Actually falling back to dmd is also ok.